### DISCUSSION TITLE
I tried change the version, and the way to use regions plugins, my apps always says  "VM4703 app.js:67 Uncaught (in promise) TypeError: wavesurfer.enableDragSelection is not a function"

### DISCUSSION THREAD

**Discussion by gylgylgyl123** [katspaugh/wavesurfer.js#3386] [Q&A]
I tried change the version, and the way to use regions plugins, my apps always says  "VM4703 app.js:67 Uncaught (in promise) TypeError: wavesurfer.enableDragSelection is not a function" 
in my html files :
`    <script src="https://unpkg.com/wavesurfer.js@7"></script>
    <script src="https://unpkg.com/wavesurfer.js@7/dist/plugins/regions.min.js"></script>
    <script src="{% static 'app.js' %}"></script> 
src="https://github.com/katspaugh/wavesurfer.js/assets/65530993/b76f48f6-ada6-4212-80e5-58d91f1fa50a">
`

and in my js files:
let wavesurfer;

function uploadAudio() {
    const input = document.getElementById('audioInput');
    const file = input.files[0];

    if (file) {
        const reader = new FileReader();

        reader.onload = function (e) {
            const audioData = e.target.result;
            initWaveform(audioData);
        };

        reader.readAsArrayBuffer(file);
    }
}

function initWaveform(audioData) {
    wavesurfer = WaveSurfer.create({
        container: '#waveform',
        waveColor: 'violet',
        progressColor: 'purple'
    });

    // wavesurfer = WaveSurfer.create({
    //     container: '#waveform',    // 指定 WaveSurfer 绑定的元素
    //     waveColor: 'violet',       // 波形的颜色
    //     progressColor: 'purple',   // 播放进度的颜色
    //     plugins: [
    //         WaveSurfer.regions.create({
    //             // 这里可以添加 regions 插件的特定配置
    //             // 例如，定义区域的颜色、拖动限制等
    //             dragSelection: true,    // 启用拖动以创建新区域
    //             color: 'rgba(0, 255, 0, 0.3)'  // 新创建的区域的颜色
    //         })
    //     ]
    // });

    // var wavesurfer = WaveSurfer.create({
    //     container: '#waveform',    // 指定 WaveSurfer 绑定的元素
    //     waveColor: 'violet',       // 波形的颜色
    //     progressColor: 'purple',   // 播放进度的颜色
    //     plugins: [
    //         WaveSurfer.regions.create({
    //             regions: [
    //                 {
    //                     start: 1,
    //                     end: 3,
    //                     color: 'hsla(400, 100%, 30%, 0.5)'
    //                 }, {
    //                     start: 5,
    //                     end: 7,
    //                     color: 'hsla(200, 50%, 70%, 0.4)'
    //                 }
    //             ],
    //             dragSelection: {
    //                 slop: 5
    //             }
    //         })
    //     ]
    // });

    wavesurfer.loadBlob(new Blob([audioData]));

    wavesurfer.on('ready', function () {
        wavesurfer.enableDragSelection({});
    });
}

<img width="1280" alt="Snipaste_2023-12-07_23-07-55" src="https://github.com/katspaugh/wavesurfer.js/assets/65530993/e0b69706-c32c-46fb-94bf-eb7d15cb64fc">

**Comment by katspaugh** [comment]
It should be `regionsPlugin.enableDragSelection()`. Check out this example: https://wavesurfer.xyz/examples/?regions.js
